JP-3649 Fix cube build error when called from mrs_imatch and add deprecated warning for mrs_imatch (#8728)
spacetelescope/jwst: docs/jwst/mrs_imatch/description.rst

Overview
--------

> This step has been deprecated in JWST calibration pipeline version 11.1.

Reviewer: Sorry, I missed this earlier.  The build will be 11.1, the pipeline version is not yet known.  Maybe we should revise this to just say that the step is deprecated and will be skipped by default?  They can refer to the change log if they need to know when it became deprecated.

Author: done

> The mrs_imatch step is skipped in the default running of the calwebb_spec3 pipeline.

Reviewer: ```suggestion
The ``mrs_imatch`` step is skipped by default in the calwebb_spec3 pipeline.
```

Author: done
